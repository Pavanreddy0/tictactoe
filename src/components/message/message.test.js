import { render, screen, cleanup } from "@testing-library/react";
import Message from "./message";
import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";

afterEach(cleanup);

it("button renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<Message />, div);
});

it("button renders successfully", () => {
  const { getByTestId } = render(<Message value="This is Pavan" />);
  expect(getByTestId("button")).toHaveTextContent("This is Pavan");
});

it("matches snapshow", () => {
  const tree = renderer.create(<Message value="Hi"></Message>).toJSON();
  expect(tree).toMatchSnapshot();
});

it("matches snapshot 2", () => {
  const tree = renderer.create(<Message value="Pavan"></Message>).toJSON();
  expect(tree).toMatchSnapshot();
});
