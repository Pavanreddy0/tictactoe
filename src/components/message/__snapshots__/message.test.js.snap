// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 2 1`] = `
<div
  data-testid="button"
>
  <h1
    name="msg"
    style={
      Object {
        "display": "grid",
        "fontSize": "20px",
        "fontWeight": "800",
        "margin": "10px auto",
        "textAlign": "center",
        "width": "300px",
      }
    }
  >
    Pavan
  </h1>
</div>
`;

exports[`matches snapshow 1`] = `
<div
  data-testid="button"
>
  <h1
    name="msg"
    style={
      Object {
        "display": "grid",
        "fontSize": "20px",
        "fontWeight": "800",
        "margin": "10px auto",
        "textAlign": "center",
        "width": "300px",
      }
    }
  >
    Hi
  </h1>
</div>
`;
